test/extended/util/oauthserver.go

```diff
@@ -0,0 +1,590 @@
+	legacyconfigv1 "github.com/openshift/api/legacyconfig/v1"
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	osinv1 "github.com/openshift/api/osin/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	oauthclient "github.com/openshift/client-go/oauth/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/helpers"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/oc/pkg/helpers/tokencmd"
+	"github.com/openshift/origin/test/extended/testdata"
+)
+
+const (
+	serviceURLFmt = "https://test-oauth-svc.%s.svc" // fill in the namespace
+
+	servingCertDirPath  = "/var/config/system/secrets/serving-cert"
+	servingCertPathCert = "/var/config/system/secrets/serving-cert/tls.crt"
+	servingCertPathKey  = "/var/config/system/secrets/serving-cert/tls.key"
+
+	routerCertsDirPath = "/var/config/system/secrets/router-certs"
+
+	sessionSecretDirPath = "/var/config/system/secrets/session-secret"
+	sessionSecretPath    = "/var/config/system/secrets/session-secret/session"
+
+	oauthConfigPath  = "/var/config/system/configmaps/oauth-config"
+	serviceCADirPath = "/var/config/system/configmaps/service-ca"
+
+	configObjectsDir = "/var/oauth/configobjects/"
+
+	RouteName = "test-oauth-route"
+	SAName    = "e2e-oauth"
+)
+
+var (
+	serviceCAPath = "/var/config/system/configmaps/service-ca/service-ca.crt" // has to be var so that we can use its address
+
+	osinScheme = runtime.NewScheme()
+	codecs     = serializer.NewCodecFactory(osinScheme)
+	encoder    = codecs.LegacyCodec(osinv1.GroupVersion)
+
+	defaultProcMount         = corev1.DefaultProcMount
+	volumesDefaultMode int32 = 420
+)
+
+func init() {
+	utilruntime.Must(osinv1.Install(osinScheme))
+}
+
+func RequestTokenForUser(reqOpts *tokencmd.RequestTokenOptions, username, password string) (string, error) {
+	reqOpts.Handler = &tokencmd.BasicChallengeHandler{
+		Host:     reqOpts.ClientConfig.Host,
+		Username: username,
+		Password: password,
+	}
+
+	return reqOpts.RequestToken()
+}
+
+// DeployOAuthServer - deployes an instance of an OpenShift OAuth server
+// very simplified for now
+// returns OAuth server url, cleanup function, error
+func DeployOAuthServer(oc *CLI, idps []osinv1.IdentityProvider, configMaps []corev1.ConfigMap, secrets []corev1.Secret) (*tokencmd.RequestTokenOptions, func(), error) {
+	oauthServerDataDir := FixturePath("testdata", "oauthserver")
+	cleanups := func() {
+		oc.AsAdmin().Run("delete").Args("clusterrolebinding", oc.Namespace()).Execute()
+		oc.AsAdmin().Run("delete").Args("oauthclient", oc.Namespace()).Execute()
+	}
+
+	if err := oc.AsAdmin().Run("create").Args("-f", path.Join(oauthServerDataDir, "oauth-sa.yaml")).Execute(); err != nil {
+		return nil, cleanups, err
+	}
+
+	// the oauth server needs access to kube-system configmaps/extension-apiserver-authentication
+	oauthSARolebinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: oc.Namespace(), // TODO: probably something more cleaver?
```

Comment:
this is fine i think

Reply:
Ok, I'll remove the comment then. What a typo tho.